File: frontend/src/types/resource.js
/**
 * @typedef {"data_science"|"web_development"|"ai_ml"} Domain
 * @typedef {"beginner"|"intermediate"|"advanced"} ResourceLevel
 * @typedef {"course"|"article"|"video"|"project"|"assessment"|"book"} ResourceType
 *
 * @typedef {Object} Resource
 * @property {string} id
 * @property {string} title
 * @property {Domain} domain
 * @property {ResourceLevel} level
 * @property {ResourceType} type
 * @property {string[]} tags
 * @property {string[]} prerequisites
 * @property {number} duration_hours
 * @property {string} url
 * @property {string} description
 */
export {};

File: backend/app/models/resource.py
"""backend/app/models/resource.py — Pydantic model mirroring contracts/resource.schema.json"""

from __future__ import annotations

from enum import Enum

from pydantic import BaseModel


class Domain(str, Enum):
    data_science = "data_science"
    web_development = "web_development"
    ai_ml = "ai_ml"


class ResourceLevel(str, Enum):
    beginner = "beginner"
    intermediate = "intermediate"
    advanced = "advanced"


class ResourceType(str, Enum):
    course = "course"
    article = "article"
    video = "video"
    project = "project"
    assessment = "assessment"
    book = "book"


class Resource(BaseModel):
    id: str
    title: str
    domain: Domain
    level: ResourceLevel
    type: ResourceType
    tags: list[str]
    prerequisites: list[str]
    duration_hours: float
    url: str
    description: str
